health.go

```diff
@@ -80,8 +80,18 @@ type Health interface {
 	// DeregisterAll Deregister removes all health checks from this instance, and stops their next executions.
 	// It is equivalent of calling Deregister() for each currently registered check.
 	DeregisterAll()
-	// WithLogger allows you to change the logging implementation, defaults to standard logging
-	WithLogger(logger log.Logger)
+	// WithCheckListener allows you to listen to check start/end events
+	WithCheckListener(listener CheckListener)
+}
+
+// CheckListener can be used to gain check stats or log check transitions
+type CheckListener interface {
+	// CheckStarted is called when a check with the specified name has started
+	CheckStarted(name string)
+
+	// CheckCompleted is called when the check with the specified name has completed it's execution.
+	// The results are passed as an argument
+	CheckCompleted(name string, result Result)
```

Comment:
Consider rename: `OnCheckComplete`

Reply:
Do you think it's worth adding `OnCheckFailure`? I suppose that's the most interesting use case so you can support it explicitly

Reply:
Or another option: `OnCheckStatusChange`

Reply:
I decided to go for the smallest API, These functions contain all the data you need as a hook for external monitoring, logging, and to act upon status changes